main/commands/all/api/rules_add.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"fmt"
+
+	routerService "github.com/xtls/xray-core/app/router/command"
+	cserial "github.com/xtls/xray-core/common/serial"
+	"github.com/xtls/xray-core/infra/conf"
+	"github.com/xtls/xray-core/infra/conf/serial"
+	"github.com/xtls/xray-core/main/commands/base"
+)
+
+var cmdAddRules = &base.Command{
+	CustomFlags: true,
+	UsageLine:   "{{.Exec}} api adrules [--server=127.0.0.1:8080] <c1.json> [c2.json]...",
+	Short:       "Add routing rules",
+	Long: `
+Add routing rules to Xray.
+Arguments:
+	-s, -server 
+		The API server address. Default 127.0.0.1:8080
+	-t, -timeout
+		Timeout seconds to call API. Default 3
+Example:
+    {{.Exec}} {{.LongName}} --server=127.0.0.1:8080 c1.json c2.json
+`,
+	Run: executeAddRules,
+}
+
+func executeAddRules(cmd *base.Command, args []string) {
+	setSharedFlags(cmd)
+	cmd.Flag.Parse(args)
+	unnamedArgs := cmd.Flag.Args()
+	if len(unnamedArgs) == 0 {
+		fmt.Println("reading from stdin:")
+		unnamedArgs = []string{"stdin:"}
+	}
+	conn, ctx, close := dialAPIServer()
+	defer close()
+
+	client := routerService.NewRoutingServiceClient(conn)
+
+	rcs := make([]conf.RouterConfig, 0)
+	for _, arg := range unnamedArgs {
+		r, err := loadArg(arg)
+		if err != nil {
+			base.Fatalf("failed to load %s: %s", arg, err)
+		}
+		conf, err := serial.DecodeJSONConfig(r)
+		if err != nil {
+			base.Fatalf("failed to decode %s: %s", arg, err)
+		}
+		rcs = append(rcs, *conf.RouterConfig)
+	}
+	if len(rcs) == 0 {
+		base.Fatalf("no valid rule found")
+	}
+	for _, in := range rcs {
```

Comment:
Try to make this change to one call with the list of configs

Reply:
you mean i should check if any configs entered (length  of unnamedArgs)? 

https://github.com/XTLS/Xray-core/blob/6127251547685707869c9f2b86256239fd539c1e/main/commands/all/api/rules_add.go#L44-L45